pageserver/src/bin/pageserver.rs

```diff
@@ -40,6 +42,14 @@ struct CfgFileParams {
     pg_distrib_dir: Option<String>,
     auth_validation_public_key_path: Option<String>,
     auth_type: Option<String>,
+
+    relish_storage_local_path: Option<String>,
```

Comment:
This all looks pretty ugly due to two reasons:

* not that many ways to express "either single FS parameter or 2 mandatory + 2 optional S3 parameters"
* the whole fact that we have to use such parameters in our app: for instance, Amazon has `~/.aws/` dir with all that we need, for the user interface we might want to be ore user-friendly and detect such things for user instead of asking him to enter them
* local fs storage is a bit of a weird bullet itself, considering that we store our relishes locally already

Reply:
I've decided to leave out the "storage_backend=s3|localfs" param, since I already have set the relations between the new params, so you presumably cannot really mix the parameters in the wrong way.

Reply:
https://github.com/alexcrichton/toml-rs/blob/master/examples/enum_external.rs

I'll try this way and push something when ready.
This is a bit different form sections, but also nice.